### PR TITLE
Add more options for key system configuration

### PR DESCRIPTION
**Pull request by arunshan**
### This PR will...
Will add flexibility to add more configuration for key system configuration.

### Why is this Pull Request needed?
This PR is needed to support Playready DRM on PS5 platform. The keySystem should have `distinctiveIdentifier` and videoCapabilities should not contain robustness field.

Also we need to support UTF-8 encoding. We need a way to return the keyMessageXML if this is UTF-8 encoding.

### Are there any points in the code the reviewer needs to double check?
Please check buildKeySystemConfiguration and make sure it does not break existing platforms that support DRM. Especially check if `robustness` as empty string is a requirement. I am not including robustness key at all if it is not being passed in drmSystemOptions.

### Resolves issues:
Supports PS5 DRM

### CODE REVIEW

File: src/controller/eme-controller.js [hunk 1]
```diff
@@ -545,12 +546,18 @@ class EMEController extends EventHandler {
     let keyMessageXml;
 
     switch (keysListItem.mediaKeySystemDomain) {
+    // eslint-disable-next-line no-case-declarations
     case KeySystems.PLAYREADY:
+      const utfEncodingType = this._drmSystemOptions.utfEncodingType;
+      const isUtf8EncodedPlatform = utfEncodingType === 'utf-8';
+      const uIntArray = isUtf8EncodedPlatform ? new Uint8Array(keyMessage) : new Uint16Array(keyMessage);
```
Comment (springuper) [inline src/controller/eme-controller.js:543]: ok, so if we don't set `utfEncodingType`, it's still the same as before, which means we don't need to add this option for other platforms using Hls.js.

File: src/controller/eme-controller.js [hunk 2]
```diff
@@ -14,78 +14,79 @@ const { XMLHttpRequest } = window;
...
-    ]
+    // audioCapabilities: [
+    // { contentType: 'audio/mp4; codecs="mp4a.40.2"' }
+    // ],
+    // videoCapabilities: [
+    // { contentType: 'video/mp4; codecs="avc1.42E01E"' }
+    // ]
   };
 
+  if (drmSystemOptions.distinctiveIdentifier) {
```
Comment (springuper) [inline src/controller/eme-controller.js:39]: or, we can extract `robustness` property and put the rest properties on `baseConfig`:

```js
const { audioRobustness, videoRobustness, ...others } = drmSystemOptions;
baseConfig = { ...baseConfig, ...others };
...
```

File: src/controller/eme-controller.js [hunk 3]
```diff
@@ -14,78 +14,79 @@ const { XMLHttpRequest } = window;
...
-      robustness: drmSystemOptions.audioRobustness || ''
-    });
-  });
-  videoCodecs.forEach((codec) => {
-    baseConfig.videoCapabilities.push({
-      contentType: `video/mp4; codecs="${codec}"`,
-      robustness: drmSystemOptions.videoRobustness || ''
-    });
-  });
+const createWidevineMediaKeySystemConfigurations = function (audioCodecs, videoCodecs, drmSystemOptions = {}) { /* jshint ignore:line */
```
Comment (springuper) [inline src/controller/eme-controller.js]: should we remove these methods and directly use `buildKeySystemConfigurations` in https://github.com/Tubitv/hls.js/pull/8/files#diff-1cb8b36ed42a658ab63ac970ab1bd648d67f9b199f95fd279f8a50c8d51df2a6R107?

**Review by springuper — APPROVED** [review]
LGTM.

left some small comments.

File: src/controller/eme-controller.js (hunk 3 above)
Comment (mr-pinzhang) [reply src/controller/eme-controller.js]: +1, it has no differences between these 2 key systems

**Review by mr-pinzhang — APPROVED** [review]
LGTM, only left a comment that maybe we can remove `createWidevine...` or `createPlayready` and use the same function to get key system configs.

**Comment by mr-pinzhang** [comment]
after merging, then we'll need to publish the next version, version number is `v0.10.9` @arunshan

File: src/controller/eme-controller.js (hunk 1 above)
Comment (arunshan) [reply src/controller/eme-controller.js:543]: Yes thats correct.

File: src/controller/eme-controller.js (hunk 3 above)
Comment (arunshan) [reply src/controller/eme-controller.js]: Done

File: src/controller/eme-controller.js (hunk 2 above)
Comment (arunshan) [reply src/controller/eme-controller.js:39]: For PS5 we cannot have robustness setting set at all if it is not being used. So the way I have is fine IMO.

**Pull request merged**